deepspeed/utils/tensor_fragment.py

```diff
@@ -223,6 +252,22 @@ def safe_get_local_grad(param):
     return None
 
 
+def safe_set_local_grad(param, value):
+    """Update the gradient of a partitioned parameter.
+        Args:
+            param (``torch.nn.Parameter``): A model parameter
+            value (``torch.Tensor``): New value
+    """
+    if param.grad is not None:
```

Comment:
Does this condition cover all Z0,1,2? We have the same condition in `safe_set_full_grad` but it covers only Z0 there?

Reply:
Yes, there might be some inconsistency here. The `_local_` APIs were contributed by a user because of their needs, and I didn't spend too much time thinking about it. The `_local_` APIs were created specifically for Z3 since that is the case where each rank holds an equal-sized partition of every parameter/gradient. So, it might be better to throw an assert for all other cases to provide user clarity.